src/finiteVolume/processes/twoPhaseEulerAdvection.cpp

```diff
@@ -203,29 +203,103 @@ ablate::finiteVolume::processes::TwoPhaseEulerAdvection::TwoPhaseEulerAdvection(
+
     // Create the decoder based upon the eoses
     decoder = CreateTwoPhaseDecoder(flow.GetSubDomain().GetDimensions(), eosGas, eosLiquid);
 
     // Currently, no option for species advection
-    flow.RegisterRHSFunction(CompressibleFlowComputeEulerFlux, this, "euler", {"densityVF", "euler"}, {});
-    flow.RegisterRHSFunction(CompressibleFlowComputeVFFlux, this, "densityVF", {"densityVF", "euler"}, {});
+    flow.RegisterRHSFunction(CompressibleFlowComputeEulerFlux, this, "euler", {"volumeFraction", "densityVF", "euler"}, {});
+    flow.RegisterRHSFunction(CompressibleFlowComputeVFFlux, this, "densityVF", {"volumeFraction", "densityVF", "euler"}, {});
 
     // check to see if auxFieldUpdates needed to be added
     if (flow.GetSubDomain().ContainsField(CompressibleFlowFields::VELOCITY_FIELD)) {
         flow.RegisterAuxFieldUpdate(UpdateAuxVelocityField2Gas, nullptr, std::vector<std::string>{CompressibleFlowFields::VELOCITY_FIELD}, {CompressibleFlowFields::EULER_FIELD});
     }
     if (flow.GetSubDomain().ContainsField(CompressibleFlowFields::TEMPERATURE_FIELD)) {
         // add in aux update variables
-        flow.RegisterAuxFieldUpdate(UpdateAuxTemperatureField2Gas, this, std::vector<std::string>{CompressibleFlowFields::TEMPERATURE_FIELD}, {"densityVF", "euler"});
+        flow.RegisterAuxFieldUpdate(UpdateAuxTemperatureField2Gas, this, std::vector<std::string>{CompressibleFlowFields::TEMPERATURE_FIELD}, {"volumeFraction", "densityVF", "euler"});
     }
     if (flow.GetSubDomain().ContainsField("pressure")) {
         // add in aux update variables
-        flow.RegisterAuxFieldUpdate(UpdateAuxPressureField2Gas, this, std::vector<std::string>{"pressure"}, {"densityVF", "euler"});
+        flow.RegisterAuxFieldUpdate(UpdateAuxPressureField2Gas, this, std::vector<std::string>{"pressure"}, {"volumeFraction", "densityVF", "euler"});
+    }
+}
+PetscErrorCode ablate::finiteVolume::processes::TwoPhaseEulerAdvection::MultiphaseFlowPreStage(TS flowTs, ablate::solver::Solver &solver, PetscReal stagetime) {
+    PetscFunctionBegin;
+    // Get flow field data
+    const auto &fvSolver = dynamic_cast<ablate::finiteVolume::FiniteVolumeSolver &>(solver);
+    solver::Range cellRange;
+    fvSolver.GetCellRangeWithoutGhost(cellRange);
+    PetscInt dim;
+    PetscCall(DMGetDimension(fvSolver.GetSubDomain().GetDM(), &dim));
+    const auto &flowEulerId = fvSolver.GetSubDomain().GetField("euler").id;  // need this to get uOff
+    const auto &vfEulerId = fvSolver.GetSubDomain().GetField("volumeFraction").id;
+    const auto &rhoAlphaEulerId = fvSolver.GetSubDomain().GetField("densityVF").id;
+
+    DM dm = fvSolver.GetSubDomain().GetDM();
+    Vec globFlowVec;
+    PetscCall(TSGetSolution(flowTs, &globFlowVec));
+
+    PetscScalar *flowArray;
+    PetscCall(VecGetArray(globFlowVec, &flowArray));
+
+    // reproducing uOff, based on fieldId
+    PetscInt rhoAlphaId = 0, vfId = 0;
+    if (rhoAlphaEulerId > flowEulerId) {
```

Comment:
These id orders are not always fixed and rhoAlphaEulerId may be before "Euler".  You should just pull the offset value directly from the field.

`PetscInt uOff[3] = {fvSolver.GetSubDomain().GetField("euler").offset, ..., ... }`